### PR TITLE
feat(analysis): support OpenSees floor load transfer

### PR DESCRIPTION
**Pull request by qinsz01**
## Summary

- Problem: OpenSees analysis only consumed explicit `load_cases`; StructureModel V2 story `floor_loads` were not transferred into analysis actions.
- Why it matters: floor area loads are a normal building input, and users need to know whether the analysis used tributary, one-way slab, or two-way slab transfer.
- What changed: added OpenSees floor-load expansion/transfer, optional enum method selection, method summaries in tool results, and frontend visualization of original floor area loads.
- What did NOT change (scope boundary): generic model-building behavior was left unchanged; the frontend displays original story area loads separately from equivalent line/nodal loads to avoid mixing concepts.

## Change Type (select all)

- [ ] Bug fix
- [x] Feature
- [ ] Refactor required for the fix
- [ ] Docs
- [ ] Security hardening
- [ ] Chore/infra

## Scope (select all touched areas)

- [x] Gateway / orchestration
- [x] Skills / tool execution
- [ ] Auth / tokens
- [ ] Memory / storage
- [ ] Integrations
- [x] API / contracts
- [x] UI / DX
- [ ] CI/CD / infra

## Linked Issue/PR

- Closes #
- Related #
- This PR fixes a bug or regression: No

## Root Cause / Regression History (if applicable)

N/A; this is a feature addition.

- Root cause: N/A
- Missing detection / guardrail: N/A
- Prior context (`git blame`, prior PR, issue, or refactor if known): N/A
- Why this regressed now: N/A
- If unknown, what was ruled out: N/A

## Regression Test Plan (if applicable)

N/A for regression history; added feature coverage for the new behavior.

- Coverage level that should have caught this:
  - [x] Unit test
  - [x] Seam / integration test
  - [ ] End-to-end test
  - [ ] Existing coverage already sufficient
- Target test or file: `backend/src/agent-skills/analysis/opensees-static/__tests__/test_floor_load_expansion.py`, `backend/src/agent-langgraph/__tests__/analysis-tool-summary.test.mjs`, `frontend/tests/components/visualization-adapter.test.ts`
- Scenario the test should lock in: OpenSees expands floor area loads using the selected transfer method and the UI keeps original floor loads visible as area markers.
- Why this is the smallest reliable guardrail: it checks deterministic transfer math and snapshot shape without requiring a browser E2E path.
- Existing test that already covers this (if any): `node tests/runner.mjs analysis-regression` covers OpenSees routing and analysis contracts.
- If no new test is added, why not: N/A

## User-visible / Behavior Changes

- OpenSees static analysis now supports story floor area loads from StructureModel V2.
- Analysis tool output reports the floor-load transfer method and per-story transfer summary.
- Visualization shows original story floor area loads as translucent area panels with downward arrows, labeled separately from nodal/distributed load actions.
- Added bilingual UI strings for original floor area load details.

## Diagram (if applicable)

```text
Before:
[stories.floor_loads] -> [not expanded for OpenSees] -> [analysis omits floor area load]

After:
[stories.floor_loads] -> [CN auto/tributary/one-way/two-way transfer] -> [OpenSees equivalent loads]
                     -> [visualization shows original floor area load panels]
```

## Security Impact (required)

- New permissions/capabilities? (`Yes/No`): No
- Secrets/tokens handling changed? (`Yes/No`): No
- New/changed network calls? (`Yes/No`): No
- Command/tool execution surface changed? (`Yes/No`): Yes
- Data access scope changed? (`Yes/No`): No
- If any `Yes`, explain risk + mitigation: the analysis tool accepts a new optional `floorLoadTransferMode` enum. It is constrained to known values and does not introduce arbitrary command execution or file access.

### Environment

- OS: Windows / PowerShell
- Runtime/container: local Node/npm + backend Python/OpenSees runtime
- Model/provider: N/A
- Integration/channel (if any): GitHub PR from fork branch
- Relevant config (redacted): local LLM key was present for the first backend test run; reran backend tests with `LLM_API_KEY` removed from the test process.

### Steps

1. Build and type-check frontend/backend.
2. Run backend Jest with the local LLM key removed from the test process.
3. Run frontend unit tests and OpenSees analysis regression.

### Expected

- OpenSees analysis expands story floor area loads and reports transfer method details.
- Frontend visualization exposes original floor area loads separately from equivalent load actions.
- Test suites pass.

### Actual

- Expected behavior observed in tests; all final verification commands passed.

## Evidence

- `Remove-Item Env:LLM_API_KEY -ErrorAction SilentlyContinue; npm test --prefix backend -- --runInBand` passed: 81 passed, 1 skipped; 857 passed, 1 skipped.
- `node tests/runner.mjs analysis-regression` passed all OpenSees/runtime/contract checks.
- `npm run type-check --prefix frontend` passed.
- `npm run lint --prefix frontend` passed.
- `npm run test:run --prefix frontend` passed: 26 passed, 2 skipped; 194 passed, 6 todo.

## Human Verification (required)

- Verified scenarios: backend analysis/tool tests, OpenSees regression, frontend type-check/lint/unit tests, visualization adapter floor-load snapshot behavior.
- Edge cases checked: duplicate fallback story loads are not double-counted when explicit `floor_loads` exists; area loads are labeled separately from `P`/`q` actions.
- What you did not verify: interactive browser rendering through Playwright screenshots; visual behavior is covered at snapshot/type level only.

## Review Conversations

- [x] I replied to or resolved every bot review conversation I addressed in this PR.
- [x] I left unresolved only the conversations that still need reviewer or maintainer judgment.

If a bot review conversation is addressed by this PR, resolve that conversation yourself. Do not leave bot review conversation cleanup for maintainers.

## Compatibility / Migration

- Backward compatible? (`Yes/No`): Yes
- Config/env changes? (`Yes/No`): No
- Migration needed? (`Yes/No`): No
- If yes, exact upgrade steps: N/A

## Risks and Mitigations

- Risk: floor-load transfer assumptions may differ from a user's intended slab behavior.
  - Mitigation: method metadata is reported to users, and callers can select a specific transfer mode.
- Risk: visualization could be confused with equivalent analysis loads.
  - Mitigation: the UI labels these as original floor area loads and renders them as area panels rather than nodal/line load markers.

### CODE REVIEW

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_analysis.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def _run_3d_frame(self, parameters: Dict[str, Any], ops) -> Dict[str, Any]:
                 ops.fix(node_tag, *[int(bool(value)) for value in node.restraints])
 
         for elem in self.analyzer.model.elements:
-            if elem.type == 'beam':
+            if elem.type in {'beam', 'column'}:
                 self._define_beam_element(elem, ops)
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_analysis.py:125]: Addressed in 7b7bbf9: the shared 3D beam/column path now reports missing sections as a frame element error instead of beam-only wording.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 2]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+            'requestedMode': requested_mode,
+            'effectiveMode': requested_mode,
+            'loadSource': 'story_floor_loads',
+            'designCode': 'GB 50010-2010(2015) 9.1.1',
+            'method': self._floor_load_transfer_method_label(requested_mode),
+            'methodEn': self._floor_load_transfer_method_label(requested_mode),
+            'methodZh': self._floor_load_transfer_method_label_zh(requested_mode),
+            'items': [],
+            'warnings': mode_warnings,
+        }
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py:1396]: Addressed in 7b7bbf9: the floor-load transfer summary now refreshes top-level effectiveMode/method from actual trace items, using the concrete mode when uniform and mixed when multiple methods are applied.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 3]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+            return slab_loads
+
+        warnings = self._floor_load_trace_warnings()
+        warnings.append('No complete supported slab panel was found; falling back to node tributary-area loads.')
+        self._floor_load_transfer_trace['warnings'] = warnings
+        self._floor_load_transfer_trace['effectiveMode'] = 'node_tributary'
+        self._floor_load_transfer_trace['method'] = self._floor_load_transfer_method_label('node_tributary')
+        self._floor_load_transfer_trace['methodEn'] = self._floor_load_transfer_method_label('node_tributary')
+        self._floor_load_transfer_trace['methodZh'] = self._floor_load_transfer_method_label_zh('node_tributary')
+        return self._expand_story_floor_loads_to_nodes(parameters, specs, 'node_tributary')
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py]: Addressed in 7b7bbf9: slab-beam transfer now falls back per story to node tributary-area loads when that story has incomplete slab support, so successful stories no longer mask missing loads on other stories. Added a mixed-support regression test.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 4]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+
+        inferred = self._infer_floor_load_type_from_case_id(load_case_id)
+        return {inferred} if inferred else None
+
+    def _infer_floor_load_type_from_case_id(self, load_case_id: str) -> Optional[str]:
+        normalized = re.sub(r'[^a-z0-9]+', '', load_case_id.lower())
+        if normalized in {'d', 'dl', 'dead', 'deadload', 'lcde'} or normalized.startswith('dead'):
+            return 'dead'
+        if normalized in {'l', 'll', 'live', 'liveload', 'lcll'} or normalized.startswith('live'):
+            return 'live'
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py:1358]: Addressed in 7b7bbf9: load-case ID inference now recognizes LC-DEAD/LC-LIVE style prefixes, while keeping the previous aliases. Added coverage for those common IDs.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 5]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+                self._append_floor_load_trace_item({
+                    'story': str(self._get_field(story, 'id', '')),
+                    'method': self._floor_load_transfer_method_label(effective_mode),
+                    'methodEn': self._floor_load_transfer_method_label(effective_mode),
+                    'methodZh': self._floor_load_transfer_method_label_zh(effective_mode),
+                    'effectiveMode': effective_mode,
+                    'loadIntensityKNPerM2': intensity,
+                    'generatedLoadType': 'nodal',
+                    'generatedLoadCount': len(self._story_floor_node_areas(story, parameters)),
+                    'totalLoadKN': intensity * sum(self._story_floor_node_areas(story, parameters).values()),
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py]: Addressed in 7b7bbf9: node tributary areas are computed once per story expansion and reused for load emission plus trace generatedLoadCount/totalLoadKN.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 6]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+            if start is None or end is None:
+                continue
+            p_start = (float(start.x), float(start.y), float(start.z))
+            p_end = (float(end.x), float(end.y), float(end.z))
+            a = (x1, y1, z)
+            b = (x2, y2, z)
+            if self._points_match(p_start, a) and self._points_match(p_end, b):
+                return str(elem.id)
+            if self._points_match(p_start, b) and self._points_match(p_end, a):
+                return str(elem.id)
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py]: Addressed in 7b7bbf9: beam lookup now accepts beams that contain the panel edge segment. For continuous beams, the generated uniform load is scaled by segment length over element length so total applied gravity remains consistent. Added a continuous-beam regression test.

File: backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py [hunk 7]
```diff
@@ -1199,57 +1219,739 @@ def _build_3d_frame_local_stiffness(
...
+            return {}
+
+        x_values = sorted({float(node.x) for node in target_nodes})
+        y_values = sorted({float(node.y) for node in target_nodes})
+        x_lengths = self._tributary_lengths(x_values)
+
+        if len(y_values) >= 2:
+            y_lengths = self._tributary_lengths(y_values)
+            return {
+                str(node.id): x_lengths.get(float(node.x), 0.0) * y_lengths.get(float(node.y), 0.0)
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/opensees-static/opensees_static_simplified_static_analysis.py]: Addressed in 7b7bbf9: tributary-length lookup now uses quantized coordinate keys instead of raw float keys, and coordinate lists are deduplicated through the same quantization.

**Review by guyi2000 — APPROVED** [review]
